Simplified/NYPLAccountSignInViewController.m

```diff
@@ -702,26 +704,80 @@ - (void)validateCredentials
   NSMutableURLRequest *const request =
     [NSMutableURLRequest requestWithURL:[NYPLConfiguration loanURL]];
   
-  // Necessary to support longer login times when using usernames.
   request.timeoutInterval = 20.0;
   
-  request.HTTPMethod = @"HEAD";
-  
   self.isCurrentlySigningIn = YES;
   NSURLSessionDataTask *const task =
     [self.session
      dataTaskWithRequest:request
-     completionHandler:^(__attribute__((unused)) NSData *data,
+     completionHandler:^(NSData *data,
                          NSURLResponse *const response,
                          NSError *const error) {
        
        self.isCurrentlySigningIn = NO;
-       // This cast is always valid according to Apple's documentation for NSHTTPURLResponse.
+
        NSInteger const statusCode = ((NSHTTPURLResponse *) response).statusCode;
        
-       // Success.
        if(statusCode == 200) {
+         
+#if defined(FEATURE_DRM_CONNECTOR)        
+
+         NYPLXML *loansXML = [NYPLXML XMLWithData:data];
+         NYPLOPDSFeed *loansFeed = [[NYPLOPDSFeed alloc] initWithXML:loansXML];
+         
+         if (loansFeed.licensor) {
+           [[NYPLAccount sharedAccount] setLicensor:loansFeed.licensor];
+         } else {
+           NYPLLOG(@"Login Failed: No Licensor Token received or parsed from OPDS Loans feed");
+           [self authorizationAttemptDidFinish:NO error:nil];
+           return;
+         }
+         
+         NSMutableArray *licensorItems = [[loansFeed.licensor[@"clientToken"] stringByReplacingOccurrencesOfString:@"\n" withString:@""] componentsSeparatedByString:@"|"].mutableCopy;
+         NSString *tokenPassword = [licensorItems lastObject];
+         [licensorItems removeLastObject];
```

Comment:
It's not obvious to me that this correct not knowing what a licensor is. We're clearly depending on the server giving us something in some particular format, so it would be great if we could parse it and validate it earlier on in this process rather than depending on it being a string with a particular format at this point. I realize this is a holdover from the earlier implementation however, so if this is the best compromise for now, I am okay with that.